### PR TITLE
Add issue repository for testing

### PR DESCRIPTION
**Pull request by pablothedude**
In order to allow local development without the need of a jira
server a repository is added to reproduce the behavior of jira.
To prevent coupling with third party dependencies a couple of domain
specific valueobjects were added to return from the repositories.

https://www.pivotaltracker.com/story/show/165993367

### CODE REVIEW

**Comment by pablothedude** [comment]
@MKodde could you test the jira repository with your local dev stack with `\Surfnet\ServiceProviderDashboard\Tests\Unit\Infrastructure\DashboardBundle\Jira\Repository\IssueRepositoryTest`. It should work for the Jira repository.

File: src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/Issue.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * @var string
+     */
+    private $issueType;
+
+    /**
+     * @param string $key
+     * @param string $issueType
+     */
+    public function __construct($key, $issueType)
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/Issue.php]: You might want to add input validation at this point!

File: src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/IssueCollection.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Domain\ValueObject;
+
+class IssueCollection implements \Countable
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/IssueCollection.php]: Even though this is a very lightweight, not complex class. I would like to see a unit test covering the public interface of this class.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/Jira/Repository/DevelopmentIssueRepository.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Copyright 2018 SURFnet B.V.
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/Jira/Repository/DevelopmentIssueRepository.php]: ~~2018~~ 2019

File: src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/Issue.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Domain\ValueObject;
+
+class Issue
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/Issue.php]: This VO should be JSON serializable, when saving this file to file in the issues.json file, it is saved as an empty object. Resulting in fatal errors on the user side.

File: src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/IssueCollection.php (hunk 2 above)
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/IssueCollection.php]: Please import the Countable

File: src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/Issue.php (hunk 4 above)
Comment (MKodde) [reply src/Surfnet/ServiceProviderDashboard/Domain/ValueObject/Issue.php]: The actual issue data was not stored in the JSON file, only the key was. I added jsonSerialize to the Issue type, this did the trick.

**Comment by pablothedude** [comment]
@MKodde thanks for addressing the raised concerns. The changes look good to me, the only concern I have is the functional aspect of the jira connection because I don't have a working Jira instance in my dev stack. So if that is tested it all looks good to me.

**Comment by MKodde** [comment]
I don't have one running either, and I would have to go through a new registration procedure. I suggest testing this on the test environment before releasing this. @quartje can you live with that?

**Comment by quartje** [comment]
Sure, I can live with that @MKodde

**Pull request merged**